### DISCUSSION TITLE
how to put a copy button in a post with code

### DISCUSSION THREAD

**Discussion by blusens** [importantimport/urara#78] [Q&A] [answered]
hello, nice proyect,  Tried using sveltekit to copy the content of page.svelte.md to use a button inside ``` python
```so that the user can copy the content of the code but at the moment they did not find how to do it, any guide?

**Comment by Interstellar750** [comment, marked as answer]
Do you mean the "Copy Button" in the upper right corner of [Fenced Code Blocks](https://www.markdownguide.org/extended-syntax/#fenced-code-blocks)?

Of course, Urara does not include this button.

Among the sites built based on Urara that I know of, there are probably two implementation methods, they are:

[kwchang0831](https://github.com/kwchang0831)'s [urara.kwchang0831.dev](https://github.com/kwchang0831/urara.kwchang0831.dev)
[Sevichecc](https://github.com/Sevichecc)'s [seviche.cc](https://github.com/Sevichecc/Urara-Blog)

You can take a look at their repositories, pick one you like, and add the required code to your repository.

But I don't know how kwchang0831 made this button, I can only tell you how to add it according to Sevichecc's method:

1. Get the [`/src/lib/utils/copyCode.ts`](https://github.com/Sevichecc/Urara-Blog/blob/main/src/lib/utils/copyCode.ts) file of the [Sevichecc/Urara-Blog](https://github.com/Sevichecc/Urara-Blog) repository, and put it into the same `/src/lib/utils/` directory in your repository.
2. Following the [`7e7745e`](https://github.com/Sevichecc/Urara-Blog/commit/7e7745ea05047b663593bbb854dc8cacc0f87348#diff-ea35660043dddc5e71c1f632be1355faaffd6ac8be86c7052144b490079a506f) commit changes in the [Sevichecc/Urara-Blog](https://github.com/Sevichecc/Urara-Blog) repository, modify `/src/lib/components/post_container.svelte` in your repository.

After that, the button should appear. If it doesn't work, try restarting the dev server or rebuilding.

Thanks for @Sevichecc. 👍
